nullaway/src/test/java/com/uber/nullaway/ArrayTests.java

```diff
@@ -0,0 +1,173 @@
+            "      o1 = fizz;",
+            "      // BUG: Diagnostic contains: dereferenced expression fizz is @Nullable",
+            "      o1 = fizz.length;",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void typeUseLegacyAnnotationOnArray() {
+    makeLegacyModeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  // ok only for backwards compat",
+            "  @Nullable Object[] foo1 = null;",
+            "  // ok according to spec",
+            "  Object @Nullable[] foo2 = null;",
+            "  // ok, but elements are not treated as @Nullable outside of JSpecify mode",
+            "  @Nullable Object @Nullable[] foo3 = null;",
+            "  // ok only for backwards compat",
+            "  @Nullable Object [][] foo4 = null;",
+            "  // ok according to spec",
+            "  Object @Nullable [][] foo5 = null;",
+            "  // ok, but @Nullable applies to first array dimension not the elements or the array ref",
+            "  Object [] @Nullable [] foo6 = null;",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void typeUseAnnotationOnArray() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  // @Nullable is not applied on top-level of array",
+            "  // BUG: Diagnostic contains: assigning @Nullable expression to @NonNull field",
+            "  @Nullable Object[] foo1 = null;",
+            "  // ok according to spec",
+            "  Object @Nullable[] foo2 = null;",
+            "  // ok according to spec",
+            "  @Nullable Object @Nullable [] foo3 = null;",
+            "  // @Nullable is not applied on top-level of array",
+            "  // BUG: Diagnostic contains: assigning @Nullable expression to @NonNull field",
+            "  @Nullable Object [][] foo4 = null;",
+            "  // ok according to spec",
+            "  Object @Nullable [][] foo5 = null;",
+            "  // @Nullable is not applied on top-level of array",
+            "  // BUG: Diagnostic contains: assigning @Nullable expression to @NonNull field",
+            "  Object [] @Nullable [] foo6 = null;",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void typeUseAndDeclarationAnnotationOnArray() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Nullable.java",
+            "package com.uber;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Target;",
+            "@Target({ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE, ElementType.TYPE_USE})",
+            "public @interface Nullable {}",
```

Comment:
Let's use two test source files for clean-ness.  Rough suggestion:
```suggestion
            "public @interface Nullable {}")
        .addSourceLines("Test.java",
            "package com.uber;",
```
Please make the change in other places as well.

Reply:
Fixed